test/test_unistd.rs

```diff
@@ -64,22 +64,28 @@ fn test_wait() {
 
 #[test]
 fn test_mkstemp() {
-    let result = mkstemp("/tmp/nix_tempfile.XXXXXX");
+    use std::env;
+
+    let mut path = env::temp_dir();
+    path.push("nix_tempfile.XXXXXX");
+
+    let result = mkstemp(&path);
     match result {
         Ok((fd, path)) => {
             close(fd).unwrap();
             unlink(path.as_path()).unwrap();
         },
         Err(e) => panic!("mkstemp failed: {}", e)
     }
+}
 
-    let result = mkstemp("/tmp/");
-    match result {
-        Ok(_) => {
-            panic!("mkstemp succeeded even though it should fail (provided a directory)");
-        },
-        Err(_) => {}
-    }
+#[test]
+#[should_panic]
```

Comment:
Our test shouldn't panic, you should just not use `unwrap()` here. You should use assertions to check the return value.